packages/adaptive_scaffold/lib/src/adaptive_scaffold.dart

```diff
@@ -0,0 +1,494 @@
+  /// a list of [NavigationDestination]s.
+  static BottomNavigationBar toBottomNavigationBar(
+      {required List<NavigationDestination> destinations}) {
+    return BottomNavigationBar(
+      items: destinations
+          .map((NavigationDestination e) => _toBottomNavItem(e))
+          .toList(),
+    );
+  }
+
+  /// Animation from bottom offscreen up onto the screen.
+  static AnimatedWidget bottomToTop(Widget child, Animation<double> animation) {
+    return SlideTransition(
+      position: Tween<Offset>(
+        begin: const Offset(0, 1),
+        end: Offset.zero,
+      ).animate(animation),
+      child: child,
+    );
+  }
+
+  /// Animation from on the screen down off the screen.
+  static AnimatedWidget topToBottom(Widget child, Animation<double> animation) {
+    return SlideTransition(
+      position: Tween<Offset>(
+        begin: Offset.zero,
+        end: const Offset(0, 1),
+      ).animate(animation),
+      child: child,
+    );
+  }
+
+  /// Animation from left off the screen into the screen.
+  static AnimatedWidget leftOutIn(Widget child, Animation<double> animation) {
+    return SlideTransition(
+      position: Tween<Offset>(
+        begin: const Offset(-1, 0),
+        end: Offset.zero,
+      ).animate(animation),
+      child: child,
+    );
+  }
+
+  /// Animation from on screen to left off screen.
+  static AnimatedWidget leftInOut(Widget child, Animation<double> animation) {
+    return SlideTransition(
+      position: Tween<Offset>(
+        begin: Offset.zero,
+        end: const Offset(-1, 0),
+      ).animate(animation),
+      child: child,
+    );
+  }
+
+  /// Animation from right off screen to on screen.
+  static AnimatedWidget rightOutIn(Widget child, Animation<double> animation) {
+    return SlideTransition(
+      position: Tween<Offset>(
+        begin: const Offset(1, 0),
+        end: Offset.zero,
+      ).animate(animation),
+      child: child,
+    );
+  }
+
+  /// Fade in animation.
+  static Widget fadeIn(Widget child, Animation<double> animation) {
+    return FadeTransition(
+      opacity: CurvedAnimation(parent: animation, curve: Curves.easeInCubic),
+      child: child,
+    );
+  }
+
+  /// Fade out animation.
+  static Widget fadeOut(Widget child, Animation<double> animation) {
+    return FadeTransition(
+      opacity: CurvedAnimation(
+          parent: ReverseAnimation(animation), curve: Curves.easeInCubic),
+      child: child,
+    );
+  }
+
+  /// Keep widget on screen while it is leaving
+  static Widget stayOnScreen(Widget child, Animation<double> animation) {
+    return FadeTransition(
+      opacity: Tween<double>(begin: 1.0, end: 1.0).animate(animation),
+      child: child,
+    );
+  }
+
+  @override
+  State<AdaptiveScaffold> createState() => _AdaptiveScaffoldState();
+}
+
+class _AdaptiveScaffoldState extends State<AdaptiveScaffold> {
+  @override
+  Widget build(BuildContext context) {
+    return Directionality(
+      textDirection: TextDirection.ltr,
```

Comment:
@youssef-attia Is there a reason why this is here ?

Reply:
Looks like a remnant of testing. Directionality for the adaptive scaffold can be controlled in many ways and should not be set to a constant. Good catch.